pallene/checker.lua

```diff
@@ -274,16 +319,43 @@ function Checker:check_program(prog_ast)
 
             for _, tl_var in ipairs(tl_group) do
                 local loc = tl_var.loc
-                local value = tl_var.value
-                local name = tl_var.decl.name
-                local typ, exp = toplevel_fun_checker:check_initializer_exp(
-                    tl_var.decl, value,
-                    "declaration of module variable %s", name)
-                local _ = self:add_global(name, typ)
-                local var = ast.Var.Name(loc, name)
-                toplevel_fun_checker:check_var(var)
-                table.insert(toplevel_stats,
-                    ast.Stat.Assign(loc, var, exp))
+
+                local last_val = tl_var.values[#tl_var.values]
+                if last_val._tag == "ast.Exp.CallFunc" then
+                    -- print(ii(last_val))
+                    last_val = toplevel_fun_checker:check_exp_synthesize(last_val)
+                    if last_val._types and #last_val._types > 1 then
```

Comment:
1 - Makes sense, but currently `_types` not being set has a meaning (see other comments)
2 - I didn't bother because I saw that it actually doesn't, but fixed it
3 - Absolutely! I forgot to remove those from an older code I was working on, sorry!